src/v/redpanda/debug_bundle.cc

```diff
@@ -0,0 +1,510 @@
+    } else if (metric.units == "d") {
+        return fmt::format(
+          "{}d",
+          std::chrono::duration_cast<std::chrono::days>(metric.interval_ms)
+            .count());
+    } else if (metric.units == "w") {
+        return fmt::format(
+          "{}w",
+          std::chrono::duration_cast<std::chrono::weeks>(metric.interval_ms)
+            .count());
+    } else if (metric.units == "y") {
+        return fmt::format(
+          "{}y",
+          std::chrono::duration_cast<std::chrono::years>(metric.interval_ms)
+            .count());
+    } else {
+        vlog(
+          bundle_log.warn,
+          "Unknown units {} for metrics-interval, defaulting to ms");
+        return fmt::format("{}ms", metric.interval_ms.count());
+    }
+}
+
+rpk_debug_bundle::rpk_debug_bundle()
+  : process{nullptr}
+  , control{ss::make_ready_future<>()}
+  , creds{std::nullopt} {}
+
+rpk_debug_bundle::rpk_debug_bundle(
+  ss::sstring host_path,
+  ss::sstring host_home,
+  const std::filesystem::path write_dir,
+  const std::filesystem::path rpk_path,
+  const debug_bundle_params params,
+  std::optional<debug_bundle_credentials> creds)
+  : process{nullptr}
+  , control{ss::make_ready_future<>()}
+  , host_path{host_path}
+  , host_home{host_home}
+  , write_dir{write_dir}
+  , rpk_path{rpk_path}
+  , params{std::move(params)}
+  , creds{std::move(creds)} {}
+
+std::vector<ss::sstring> rpk_debug_bundle::make_rpk_args() {
+    auto bundle_name = make_bundle_filename(
+      write_dir, ss::sstring{default_bundle_name});
+    std::vector<ss::sstring> rpk_argv{
+      rpk_path.string(), "debug", "bundle", "--output", bundle_name};
+
+    if (creds.has_value()) {
+        rpk_argv.push_back("--user");
+        rpk_argv.push_back(creds->username);
+        rpk_argv.push_back("--password");
+        rpk_argv.push_back(creds->password);
+        rpk_argv.push_back("--sasl-mechanism");
+        rpk_argv.push_back(creds->mechanism);
+    }
+
+    if (params.logs_since.has_value()) {
+        rpk_argv.push_back("--logs-since");
+        rpk_argv.push_back(to_journalctl_fmt(params.logs_since.value()));
+    }
+
+    if (params.logs_until.has_value()) {
+        rpk_argv.push_back("--logs-until");
+        rpk_argv.push_back(to_journalctl_fmt(params.logs_until.value()));
+    }
+
+    if (params.logs_size_limit.has_value()) {
+        rpk_argv.push_back("--logs-size-limit");
+        rpk_argv.push_back(params.logs_size_limit.value());
+    }
+
+    if (params.metrics_interval.has_value()) {
+        rpk_argv.push_back("--metrics-interval");
+        rpk_argv.push_back(to_time_fmt(params.metrics_interval.value()));
+    }
+
+    return rpk_argv;
+}
+
+template<class>
+inline constexpr bool always_false_v = false;
+
+errc rpk_debug_bundle::handle_wait_status(
+  ss::experimental::process::wait_status wstatus) {
+    return std::visit(
+      [](auto&& arg) {
+          using T = std::decay_t<decltype(arg)>;
+          if constexpr (std::is_same_v<
```

Comment:
if you use seastar's `ss::visit` helper you can get rid of all these `constexpr`s

Reply:
A lesson for me from this PR: Look at seastar docs more and use their API more that `std`

Reply:
>A lesson for me from this PR: Look at seastar docs more

Does seastar have docs? For what it's worth it's definitely not intuitive that Seastar would have this utility. I don't recall even how we discovered it. Probably reading some scylla/seastar code at one point

Reply:
> > A lesson for me from this PR: Look at seastar docs more
> 
> Does seastar have docs? For what it's worth it's definitely not intuitive that Seastar would have this utility. I don't recall even how we discovered it. Probably reading some scylla/seastar code at one point

https://docs.seastar.io/master/index.html

Reply:
cool!